### PR TITLE
[doc] Fix burn-tch doc for burn crate

### PR DESCRIPTION
**Pull request by syl20bnr**
### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Changes

Set the correct feature for docs build.

### CODE REVIEW

File: burn-core/Cargo.toml [hunk 1]
```diff
@@ -60,6 +60,7 @@ blas-netlib = ["burn-ndarray?/blas-netlib"]
 
 ndarray = ["burn-ndarray"]
 tch = ["burn-tch"]
+tch-doc-only = ["burn-tch/doc"]
```
Comment (nathanielsimard) [inline burn-core/Cargo.toml:63]: I would rename the feature flag to `doc-only` for all crates (burn-tch, burn-core and burn)

**Review by nathanielsimard — CHANGES_REQUESTED** [review]

**Comment by syl20bnr** [comment]
Switching to draft.
This requires more thoughts than expected.

**Comment by syl20bnr** [comment]
Not completely sure about this but this is an attempt to add the basis to make enabled features during docs.rs builds more explicit.

File: burn-tensor/Cargo.toml [hunk 2]
```diff
@@ -12,6 +12,10 @@ version.workspace = true
 
 [features]
 default = ["std"]
+doc = [
+  "default",
+  "experimental-named-tensor",
```
Comment (nathanielsimard) [inline burn-tensor/Cargo.toml:17]: Not sure I would include this one.

**Review by nathanielsimard — CHANGES_REQUESTED** [review]
I don't think we should include experimental features for now.

**Comment by syl20bnr** [comment]
Removed

**Review by nathanielsimard — APPROVED** [review]

**Pull request merged**